### PR TITLE
Improve path handling while testing name resolution 2.0

### PR DESCRIPTION
**Pull request by powerboat9**

### CODE REVIEW

File: gcc/testsuite/rust/compile/nr2/compile.exp [hunk 1]
```diff
@@ -44,23 +44,23 @@ namespace eval rust-nr2-ns {
...
-	    if {$test_dir == "."} {
-		set test_lbl $test
-	    } else {
-		set test_lbl "$test_dir/$test"
-            }
+        set directory [list {*}[file split $srcdir] {*}[file split $subdir]]
+	set directory [lreplace $directory end end]
+	set directory [list {*}$directory {*}$test_dir]
+        foreach test [lsort [glob -nocomplain -tails -directory [file join {*}$directory] *.rs]] {
+	    # use '/' as the path seperator here, for exclude file entries
```
Comment (CohenArthur) [inline gcc/testsuite/rust/compile/nr2/compile.exp:57]:
```suggestion
	    # use '/' as the path seperator here, to exclude file entries
```

**Review by CohenArthur — APPROVED** [review]
one nit but otherwise looks good

File: gcc/testsuite/rust/compile/nr2/compile.exp (hunk 1 above)
Comment (powerboat9) [reply gcc/testsuite/rust/compile/nr2/compile.exp:57]: Ah, "exclude file entries" is the noun. I should probably quote `exclude` to make that more clear

Comment (CohenArthur) [reply gcc/testsuite/rust/compile/nr2/compile.exp]: ah, gotcha! sorry I misread it

**Pull request merged**